### DISCUSSION TITLE
When there is not enough space either side add option to place the floating element inside the reference.

### DISCUSSION THREAD

**Discussion by OttlikG** [floating-ui/floating-ui#2884] [Ideas]
I really like using the library but I bumped into an edge case which needs fixing.
I would like to place the floating element inside the reference element when there is no available space on the screen for the floating ui because reference is takes up all / most of the screen . At the moment it places off screen and make the screen scrollable to the offscreen.

I would like to suggest to add it to the API to being able to specify both outer and inner placement more like the image is suggesting.

<img width="552" alt="floating-ui" src="https://github.com/floating-ui/floating-ui/assets/2817762/53201f69-929a-435f-8bc1-02670400f9f3">

Any feedback are welcomed.

**Comment by atomiks** [comment]
`shift({crossAxis: true})` does that.
